code/datums/traits/negative.dm

```diff
@@ -536,3 +536,19 @@
 	gain_text = "<span class='danger'>There's a lot on your mind right now.</span>"
 	lose_text = "<span class='notice'>Your mind finally feels calm.</span>"
 	medical_record_text = "Patient's mind is in a vulnerable state, and cannot recover from traumatic events."
+
+/datum/quirk/sheltered
+	name = "Sheltered"
+	desc = "You never learned galactic common."
+	value = -3
+	mob_trait = TRAIT_SHELTERED
+	gain_text = "<span class='danger'>You do not understand galactic common.</span>"
+	lose_text = "<span class='notice'>You start to put together what people are saying in galactic common.</span>"
+	medical_record_text = "Patient looks perplexed when questioned in galactic common."
+
+
+/datum/quirk/sheltered/on_process()
+	var/mob/living/carbon/human/H = quirk_holder
+	H.remove_language(/datum/language/common)
+	if(H.can_speak_in_language(/datum/language/draconic) == FALSE && H.can_speak_in_language(/datum/language/machine) == FALSE || H.can_speak_in_language(/datum/language/codespeak) == TRUE || H.can_speak_in_language(/datum/language/narsie) == TRUE || H.can_speak_in_language(/datum/language/ratvar) == TRUE)
```

Comment:
You'll have to explain whatever this thing is to me

Reply:
Hello Alex,
This if statement does the following to check if Japanese should be administered.

It checks if you have the following languages: draconic (lizardpeople) and machine (preternis)
If you have any of these two languages Japanese will not be given. Your secondary language then becomes your primary language.

If you know any of Cult languages or codespeak this check also adds japanese to you so that You have both your cult (or tot) language and Japanese.
Why this was done:  Imagine you spawning in as a cultist with this perk and being a human (or pod or even a degenerate moth). Without this check you would have common removed and only have your cult (or tot) language left over. This check prevents this situation.

Not giving the cult or tot Japanese would mean that they cannot "blend in" anymore because when they say anything they will speak in cultie or codespeak. No bueno

Reply:
wouldn't it be easier just to always add Japanese unless they got draconian or encoded? no exceptions just always. it covers all people.

Reply:
I don't see why not except this might need to be updated in the future as more languages are added (with both of our current solutions though). I'll go ahead and implement your change when I get home. It will look a lot less messy and keep Alex happy which is what we all really want anyway.

Reply:
Unresolving this so I don't forget to implement oklews suggestion.

Reply:
Progress update: I modified the code just didn't get a chance to test it. 
I test all my commits before submitting them. Should have this commit in a few hours.

Reply:
Resolved. See changes. Thank you!